### PR TITLE
feat(gatsby-plugin-image): data-srcset should be used to prevent overly eager loading on chrome

### PR DESCRIPTION
**Pull request by laurieontech**
Removed in #30221 and shouldn't have been

### CODE REVIEW

**Review by wardpeet — APPROVED** [review]

**Comment by ascorbic** [comment]
I think you mean #30221!

File: packages/gatsby-plugin-image/src/gatsby-ssr.tsx [hunk 1]
```diff
@@ -91,6 +91,11 @@ export function onRenderBody({ setHeadComponents }: RenderBodyArgs): void {
       }
       const placeholder = imageWrapper.querySelector("[data-placeholder-image]");
 
+      const sources = target.parentNode.querySelectorAll('source[data-srcset]');
+      for (const source of sources) {
+        source.setAttribute('src', source.dataset.src)
```
Comment (ascorbic) [inline packages/gatsby-plugin-image/src/gatsby-ssr.tsx]: Is this meant to ber srcset?

Comment (ascorbic) [reply packages/gatsby-plugin-image/src/gatsby-ssr.tsx]: haha

Comment (laurieontech) [reply packages/gatsby-plugin-image/src/gatsby-ssr.tsx]: Already pushed the fix :)

File: packages/gatsby-plugin-image/src/components/layout-wrapper.tsx [hunk 2]
```diff
@@ -24,9 +24,11 @@ if (hasNativeLazyLoadSupport) {
       mainImage.setAttribute('src', mainImage.dataset.src)
       mainImage.removeAttribute('data-src')
     }
-    if (mainImage.dataset.srcset) {
-      mainImage.setAttribute('srcset', mainImage.dataset.srcset)
-      mainImage.removeAttribute('data-srcset')
+
+    const sources = mainImage.parentNode.querySelectorAll('source[data-srcset]');
```
Comment (ascorbic) [inline packages/gatsby-plugin-image/src/components/layout-wrapper.tsx:28]: Won't this miss the `img` `srcset`?

Comment (ascorbic) [reply packages/gatsby-plugin-image/src/components/layout-wrapper.tsx:28]:
```suggestion
    const sources = mainImage.parentNode.querySelectorAll('[data-srcset]');
```

Comment (laurieontech) [reply packages/gatsby-plugin-image/src/components/layout-wrapper.tsx:28]: I think I shouldn't have removed the previous code. Both should exist. One will handle image and the other the additional sources.

**Review by ascorbic — DISMISSED** [review]

File: integration-tests/images/__tests__/images.ts [hunk 3]
```diff
@@ -52,7 +52,7 @@ function getImages(testId: string, dom: HTMLDivElement) {
 
   sources?.forEach(source => {
     const format = source.getAttribute("type")
-    const srcset = source.getAttribute("srcset")
+    const srcset = source.getAttribute("data-srcset")
```
Comment (ascorbic) [inline integration-tests/images/__tests__/images.ts:55]: `source.dataset.srcset` for consistency

**Pull request merged**